vllm/sequence.py

```diff
@@ -139,7 +139,18 @@ def get_output_len(self) -> int:
         return len(self.output_token_ids)
 
     def get_token_ids(self) -> List[int]:
-        return self.prompt_token_ids + self.output_token_ids
+        return list(self.prompt_token_ids) + self.output_token_ids
+
+    def get_prefix_token_ids(
+            self, num_tokens: int
+    ) -> Tuple[Tuple[int, ...], Optional[Tuple[int, ...]]]:
+        """Get prefix tokens, and make the return value hashable"""
+        prompt_length = len(self.prompt_token_ids)
+        if num_tokens > prompt_length:
```

Comment:
when does this happen?

Reply:
This happens when calculating hashes for both the user input (i.e. the prompt tokens) and the LLM-generated output (i.e. output tokens ).

Reply:
Can it happen under normal circumstance or is it only for recomputation case?

Reply:
Yes it will happen for normal cases (function `_allocate_last_physical_block` in `vllm/sequence.py`).